src/Chapter3.hs

```diff
@@ -910,6 +995,27 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = Gold { unGold :: Int }
+
+data List a
+    = Empty
+    | Cons a (List a)
+
+instance Append Gold where
+  append g1 g2 = Gold { unGold = unGold g1 + unGold g2 }
+
+instance Append (List a) where
+  append l1 l2 = appendAcc l1 l2 Empty where
+    reverseList :: List a -> List a
+    reverseList l = reverseAcc l Empty
+
+    reverseAcc :: List a -> List a -> List a
+    reverseAcc (Cons h t) acc = reverseAcc t (Cons h acc)
+    reverseAcc _                acc = acc
+
+    appendAcc (Cons h1 t1) l2                  acc = appendAcc t1 l2 (Cons h1 acc)
+    appendAcc Empty               (Cons h2 t2) acc = appendAcc Empty t2 (Cons h2 acc)
+    appendAcc _                   _            acc = reverseList acc
```

Comment:
I feel like this implementation is overcomplicated a bit 🤔 
To append two lists, you need to implement a function similar to `(++)` for lists. And it can be implemented simpler in 3 lines with straightforward pattern matching 🙂 If you have troubles implementing it, try implement it for ordinary lists manually (e.g. implement `addLists :: [a] -> [a] -> [a]` and then refactor thif function for your `List` type. You can notice that your custom list type look very similar to the built-in.

Reply:
I just found out `addLists` can be implemented in one line:
```
addLists xs ys = foldr (:) ys xs
```
(thanks to Spacemacs with the Haskell layer, whatever it's using underneath haha).